rsfbclient-core/src/charset.rs

```diff
@@ -0,0 +1,194 @@
+//! Charset definitions and functions
+//!
+//! [Reference](http://www.destructor.de/firebird/charsets.htm)
+
+use encoding::{all, types::EncodingRef, DecoderTrap, EncoderTrap};
+use std::str;
+
+use crate::FbError;
+
+/// Charset definition. Used to encode/decode the
+/// strings.
+pub struct Charset {
+    pub fb: &'static str,
+    pub str: Option<EncodingRef>,
```

Comment:
The field name `str` used in `pub str: Option<EncodingRef>` clashes with the type name `str` as in `pub fb: &'static str`.

Maybe rename for avoiding confusion or future problems with new Clippy lints, compiler warnings ?

Reply:
Done